### PR TITLE
ci: use env file instead of set_output

### PR DESCRIPTION
**Pull request by iisakkirotko**
Although the removal of set_output has been delayed, it will still likely happen at some point. See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ for details

### CODE REVIEW

**Comment by iisakkirotko** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/widgetti/solara/579?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#579** <a href="https://app.graphite.dev/github/pr/widgetti/solara/579?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#578** <a href="https://app.graphite.dev/github/pr/widgetti/solara/578?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @iisakkirotko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -112,18 +112,22 @@ jobs:
...
-        run: pip install ".[dev]" mypy==1.6.0 black==22.12.0 codespell==2.2.4 "click<8.1.4" "traitlets<5.10.0" "matplotlib<3.8.0"
+        run: |
+          mkdir -p .ci-package-locks/code-quality
+          pip install ".[dev]" mypy==1.6.0 black==22.12.0 codespell==2.2.4 "click<8.1.4" "traitlets<5.10.0" "matplotlib<3.8.0"
+          pip freeze --exclude solara --exclude solara-enterprise > ${{ env.LOCK_FILE_LOCATION }}
+          git diff --quiet || echo "HAS_DIFF=true" >> "$GITHUB_OUTPUT"
 
       - name: Install
-        if: github.event_name != 'schedule' && steps.prepare.outputs.locks_exist == 'true'
+        if: github.event_name != 'schedule' && steps.install_no_lock.outputs.LOCKS_EXIST == 'true'
```
Comment (maartenbreddels) [inline .github/workflows/test.yaml:130]:
```suggestion
        if: github.event_name != 'schedule' && steps.prepare.outputs.LOCKS_EXIST == 'true'
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -49,7 +49,7 @@ jobs:
           key: ${{ runner.os }}-js-bundle-${{ hashFiles('packages/solara-vuetify-app/**', 'packages/solara-vuetify3-app/**', 'packages/solara-widget-manager/**', 'packages/solara-widget-manager8/src/**') }}
 
       - name: Build solara widget manager
-        if: steps.cache-js-bundle-v2.outputs.cache-hit != 'true'
+        if: steps.cache-js-bundle-v2.outputs.cache-hit != true
```
Comment (maartenbreddels) [inline .github/workflows/test.yaml:52]:
```suggestion
        if: steps.cache-js-bundle-v2.outputs.cache-hit != 'true'
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -61,7 +61,7 @@ jobs:
           cd ../../
 
       - name: Build solara app package
-        if: steps.cache-js-bundle-v2.outputs.cache-hit != 'true'
+        if: steps.cache-js-bundle-v2.outputs.cache-hit != true
```
Comment (maartenbreddels) [inline .github/workflows/test.yaml:64]:
```suggestion
        if: steps.cache-js-bundle-v2.outputs.cache-hit != 'true'
```

**Review by maartenbreddels — CHANGES_REQUESTED** [review]
only errors you spotted yourself :)

**Review by maartenbreddels — APPROVED** [review]

**Comment by iisakkirotko** [comment]
### Merge activity

* **Apr 5, 8:21 AM EDT**: @iisakkirotko started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/widgetti/solara/579).

**Pull request merged**